ark/utils/notebooks_test.py

```diff
@@ -23,3 +37,234 @@ def test_example_spatial_analysis():
 
 def test_example_neighborhood_analysis():
     _exec_notebook('example_neighborhood_analysis_script.ipynb')
+
+
+# testing specific inputs for Segment_Image_Data
+def segment_notebook_setup(tb, deepcell_tiff_dir="test_tiff", deepcell_input_dir="test_input",
+                           deepcell_output_dir="test_output",
+                           single_cell_dir="test_single_cell",
+                           is_mibitiff=False, mibitiff_suffix="-MassCorrected-Filtered",
+                           num_fovs=3, num_chans=3, dtype=np.uint16):
+    # import modules and define file paths
+    tb.execute_cell('import')
+
+    # create the path to the directory containing the input data
+    tiff_path = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                             '..', '..', 'data', 'example_dataset',
+                             'input_data', deepcell_tiff_dir)
+
+    if os.path.exists(tiff_path):
+        rmtree(tiff_path)
+
+    os.mkdir(tiff_path)
+
+    # create the tif files, don't do this in notebook it's too tedious to format this
+    # also, because this is technically an input that would be created beforehand
+    if is_mibitiff:
+        fovs, chans = test_utils.gen_fov_chan_names(num_fovs=num_fovs,
+                                                    num_chans=num_chans,
+                                                    use_delimiter=True)
+        fovs = [f + mibitiff_suffix for f in fovs]
+
+        filelocs, data_xr = test_utils.create_paired_xarray_fovs(
+            tiff_path, fovs, chans, img_shape=(1024, 1024), mode='mibitiff',
+            delimiter='_', fills=False, dtype=dtype
+        )
+    else:
+        fovs, chans = test_utils.gen_fov_chan_names(num_fovs=num_fovs,
+                                                    num_chans=num_chans,
+                                                    return_imgs=False)
+
+        filelocs, data_xr = test_utils.create_paired_xarray_fovs(
+            tiff_path, fovs, chans, img_shape=(1024, 1024), delimiter='_', fills=False,
+            sub_dir="TIFs", dtype=dtype)
+
+    # define custom mibitiff paths
+    define_mibitiff_paths = """
+        base_dir = "../data/example_dataset"
+        input_dir = os.path.join(base_dir, "input_data")
+        tiff_dir = "%s/"
+        deepcell_input_dir = os.path.join(input_dir, "%s/")
+        deepcell_output_dir = os.path.join(base_dir, "%s/")
+        single_cell_dir = os.path.join(base_dir, "%s/")
+    """ % (tiff_path, deepcell_input_dir, deepcell_output_dir, single_cell_dir)
```

Comment:
What is this syntax with triple quotes followed by percent sign doing?

Reply:
It's replacing each `%s` with the corresponding variable name. So the first `%s` gets set to `tiff_path`, the second one to `deepcell_input_dir`, etc.